### PR TITLE
ethashcl doesnt depend on ethcore

### PR DESCRIPTION
**Pull request by LefterisJP**
At least not anymore. Moved global constants into ethash_cl_miner class
instead of being in Ethash class.

Also `#ifdefed` some openCL related stuff to only compile when `ETHASHCL` is active.

### CODE REVIEW

File: libethash-cl/ethash_cl_miner.h [hunk 1]
```diff
@@ -65,6 +65,14 @@ class ethash_cl_miner
 	void hash_chunk(uint8_t* _ret, uint8_t const* _header, uint64_t _nonce, unsigned _count);
 	void search_chunk(uint8_t const*_header, uint64_t _target, search_hook& _hook);
 
+	/* -- default values -- */
+	/// Default value of the local work size. Also known as workgroup size.
+	static const unsigned defaultLocalWorkSize;
```
Comment (gavofyork) [inline libethash-cl/ethash_cl_miner.h]: coding standards: should either be `c_`... if not public API or CamelCase if yes.

File: libethash-cl/ethash_cl_miner.cpp [hunk 2]
```diff
@@ -50,7 +49,10 @@
 #undef max
 
 using namespace std;
-using namespace dev::eth;
+
+const unsigned ethash_cl_miner::c_defaultLocalWorkSize = 64;
```
Comment (chfast) [inline libethash-cl/ethash_cl_miner.cpp]: I guess it should be `unsigned const`.

File: libethash-cl/ethash_cl_miner.h [hunk 3]
```diff
@@ -65,6 +65,14 @@ class ethash_cl_miner
 	void hash_chunk(uint8_t* _ret, uint8_t const* _header, uint64_t _nonce, unsigned _count);
 	void search_chunk(uint8_t const*_header, uint64_t _target, search_hook& _hook);
 
+	/* -- default values -- */
+	/// Default value of the local work size. Also known as workgroup size.
+	static const unsigned c_defaultLocalWorkSize;
```
Comment (chfast) [inline libethash-cl/ethash_cl_miner.h]: Don't want to put the values here?

Comment (LefterisJP) [inline libethash-cl/ethash_cl_miner.h]: Would produce undefined references due to being a static member of the class

**Pull request merged**